client/src/Components/TaskForm/TaskForm.js

```diff
@@ -0,0 +1,134 @@
+// Styling
+// import "./Task.css";
+import useInput from "../../hooks/useInput";
+import { useState } from "react";
+
+
+// Add verifications here
+const isNotEmpty = (value) => value.trim() !== "";
+// const isEmail = (value) => value.includes("@");
+
+// Card for general styling from UI
+import Card from "../UI/Card/Card";
+import DateSelector from "./DateSelector";
+import DropdownButton from "react-bootstrap/DropdownButton";
+import DropdownItem from "react-bootstrap/esm/DropdownItem";
+
+const TaskForm = ({ onSubmit }) => {
+
+    const {
+			value: taskTitleValue,
+			isValid: taskTitleIsValid,
+			hasError: taskTitleHasError,
+			valueChangeHandler: taskTitleChangeHandler,
+			inputBlurHandler: taskTitleBlurHandler,
+			reset: resetTaskTitle,
+		} = useInput(isNotEmpty);
```

Comment:
This is going to get very complex, with lots of aliases, as soon as you have more than one input. Maybe abstracting out your own `Input` component, that uses this hook internally, would help simplify things?

Reply:
Maybe having our own hook is not so ideal for this app, are you suggesting to managing the states individually?

Reply:
There's no problem with hooks particularly, and managing more separate states wouldn't be helpful, but imagine what happens here when you have more than one field:
```js
const {
  value: taskTitleValue,
  isValid: taskTitleIsValid,
  hasError: taskTitleHasError,
  valueChangeHandler: taskTitleChangeHandler,
  inputBlurHandler: taskTitleBlurHandler,
  reset: resetTaskTitle,
} = useInput(isNotEmpty);
const {
  value: taskDescriptionValue,
  isValid: taskDescriptionIsValid,
  hasError: taskDescriptionHasError,
  valueChangeHandler: taskDescriptionChangeHandler,
  inputBlurHandler: taskDescriptionBlurHandler,
  reset: resetTaskDescription,
} = useInput();
const {
  value: taskStatusValue,
  isValid: taskStatusIsValid,
  hasError: taskStatusHasError,
  valueChangeHandler: taskStatusChangeHandler,
  inputBlurHandler: taskStatusBlurHandler,
  reset: resetTaskStatus,
} = useInput();
```
Every one adds _another_ six variables into the current scope, another six values you have to think about when working out what the component is doing. What if instead you pushed the state down to a sub-component, instead? For example, its interface could look like:
```js
<MyInput isValid={isNotEmpty} onChange={(value) => updateForm({ ...form, title: value })} value={form.title} />
```

Reply:
makes sense, will make changes to the structure and account for the processing/creation of unnecessary variables in the scope